service-front/module/Application/src/Controller/DonorFlowController.php

```diff
@@ -157,14 +155,18 @@ public function nationalInsuranceNumberAction(): ViewModel
         $view->setVariable('form', $form);
 
         if (count($this->getRequest()->getPost())) {
-            return $this->formProcessorService->processNationalInsuranceNumberForm(
+            $formProcessorResponseDto = $this->formProcessorHellper->processNationalInsuranceNumberForm(
+                $uuid,
                 $this->getRequest()->getPost(),
                 $form,
-                $view,
                 $templates
             );
-        }
+            foreach ($formProcessorResponseDto->getVariables() as $key => $variable) {
+                $view->setVariable($key, $variable);
+            }
```

Comment:
I think you could use `setVariables` here, which would simplify the controller code quite a lot
```suggestion
            $view->setVariables($formProcessorResponseDto->getVariables());
```